### ISSUE TITLE
[2022 CVPR] PixMix: Dreamlike Pictures Comprehensively Improve Safety Measures

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#61]
<img src='https://user-images.githubusercontent.com/49643709/161482905-d892c4ab-2ecf-4ce6-91dd-5da1d09566a9.png' width=100% height=100%>
<img src='https://user-images.githubusercontent.com/49643709/161482583-1bc2da36-ded8-47c6-a6f8-41328d8f278c.png' width=50% height=50%>

Real-world machine learning applications이 reliable하고 safe systems인지 측정하는 measure들은 다음과 같음
(beyond standard test set accuracy, ML-safety measurement)

- Out-of-Distribution (OOD) robustness : robust to distribution shifts
- Prediction consistency : consistent in their predictions
- Resilence to adversaries : resilent to adversaries
- Calibrated uncertainty estimates : calibrated in their uncertainty estimates
- Ability to detect anomalous inputs : capable of identifying anomalous inputs

기존의 방법들은 performance를 improve하려면 다른 safety axes에 대해서 performance를 희생해야했음

For example, adversarial training improves adversarial robustness but sharply degrades other classifier performance metrics

Similarly, strong data augmentation and regularization techniques often improve OOD robustness but harm anomaly detection

해당 논문은 structural complexity를 이용한 새로운 data augmentation strategy를 통해 전반적인 성능을 다 높임

## 1. PIXMIX

PIXMIX는 2가지 component로 구성되어있음

Pix : a set of structurally complex pictures

Mix : pipeline for augmenting clean training pictures

High level에서 PIXMIX는 mixing set의 다양한 패턴을 training set에 통합하는 augmentation임

### 1.1. Picture Sources (PIX)
<img width="1299" alt="image" src="https://user-images.githubusercontent.com/49643709/161495388-521631c4-b3e4-4698-beff-938658b7f39f.png">

Structurally complex pictures로 fractals, feature visualizations를 사용하며, 이를 "dreamlike pictures"라고 표현

### 1.2. Mixing Pipeline (MIX)
<img width="978" alt="image" src="https://user-images.githubusercontent.com/49643709/161496648-6e54e0a2-9d7a-4e78-9de9-2a1da5da1685.png">
<img width="986" alt="image" src="https://user-images.githubusercontent.com/49643709/161497579-05f19494-a33d-4d38-8520-81f10e2035af.png">
<img width="1211" alt="image" src="https://user-images.githubusercontent.com/49643709/161497157-7eced3ff-cf6a-488d-9159-ef0cce5a1bf7.png">
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/49643709/161497257-cd0d2b8e-9ba4-4929-9f23-dec8e6482ec5.png">

### 1.3. Intuitions
> - ‘Non-accidental’ properties that humans may use, namely “structural properties of contours (orientation, length, curvature) and contour junctions (types, angles) from line drawings of natural scenes”
> - FractalDB pretrained 모델이 ImageNet-1K pretrained 모델보다 몇몇 downstream task에서 성능이 좋다는 논문
(FractalDB : algorithmically generated fractal images)
> - Self-supervised learning에서 특정 synthetic dataset이 FractalDB보다 효과가 좋다는 논문
(structural complexity and diversity are key properties for good downstream transfer)

Related works 중에 complex synthetic images로만 학습해서 잘된다는 논문들이 있었음

Structurally complex images를 data augmentation으로 활용하면 어떨까라는 궁금점에서 나옴

그리고 fractal을 사용한 이유에는 structural properties를 가지고 있으며, highly non-accidental함
(fractals posses some of these structural properties, and they are highly non-accidental and unlikely to arise from maximum entropy, unstructured random noise processes)

Gaussian noise는 input entropy를 증가시키는데 이는 descriptive complexity만 늘리지 structural complexity는 거의 없기에 다른 효과를 불러일으킴

### 2.1. CIFAR-10 / CIFAR-100
<img width="972" alt="image" src="https://user-images.githubusercontent.com/49643709/161500616-42a396b0-993d-4d4d-ac4c-4d773d2d3856.png">
<img width="959" alt="image" src="https://user-images.githubusercontent.com/49643709/161500660-cae1ca67-c079-47a0-95e4-84fa3116ffdf.png">

- Model : 40-4 Wide ResNet
- FIXMIX hyperparameter : k = 4, beta = 3
- PGD : 2/255 budget, 20 steps

### 2.2. ImageNet
<img width="953" alt="image" src="https://user-images.githubusercontent.com/49643709/161500910-3f2d6097-300a-45e7-9675-3dcef24f461a.png">
<img width="946" alt="image" src="https://user-images.githubusercontent.com/49643709/161501497-9ba913a5-c785-4688-a6c8-afd1fa8d4265.png">

- Model : pre-trained ResNet 50
- FIXMIX hyperparameter : k = 4, beta = 4
- Finetune for 90 epochs
- No adversarial results cause all tested methods accuracy declined to zero
- Since noise-based augmentations sometimes nearly overlap with the test distribution and thereby may have an unfair advantage, separately compare

### 3.1. Mixing Set
<img width="966" alt="image" src="https://user-images.githubusercontent.com/49643709/161501675-cba5c375-dff8-4595-a474-8440c587f2e4.png">

Fractals and Feature Visualization을 모두 사용하는 것이 제일 바람직함

### 3.2. Mixing Strategy
<img width="966" alt="image" src="https://user-images.githubusercontent.com/49643709/161501962-f3671be7-0fc7-4d3e-9eab-18144c623370.png">

Mix Input : only includes clean images in the mixing pipeline and does not use mixing set at all

Mix Aug : only mixes with images from the mixing set

Iterative : mixes with feature visualizations computed on the fly for the network being trained

기존 방식이 적당히 좋음

### 3.3. Hyperparameter Sensitivity
<img width="958" alt="image" src="https://user-images.githubusercontent.com/49643709/161502212-e7f56d50-e94a-4e84-998b-8fa3447d9f2d.png">

Performance is very stable across a range of hyperparameters

**Comment by Jasonlee1995** [comment]
[PixMix.pdf](https://github.com/Jasonlee1995/AI_Papers/files/8461837/PixMix.pdf)
